pepper-apis/src/main/java/org/broadinstitute/ddp/export/DataExportCoordinator.java

```diff
@@ -0,0 +1,292 @@
+            return true;
+        } catch (Exception e) {
+            LOG.error("Error while running csv export for study {}, continuing", studyGuid, e);
+            return false;
+        }
+    }
+
+    boolean exportStudyToGoogleBucket(StudyDto studyDto, DataExporter exporter, Bucket bucket,
+                                      List<ActivityExtract> activities,
+                                      Iterator<Participant> participants) {
+        try (
+                PipedOutputStream outputStream = new PipedOutputStream();
+                Writer csvWriter = new OutputStreamWriter(outputStream, StandardCharsets.UTF_8);
+                PipedInputStream csvInputStream = new PipedInputStream(outputStream, READER_BUFFER_SIZE_IN_BYTES);
+        ) {
+            // Running the DataExporter in separate thread
+            Runnable csvExportRunnable = buildExportToCsvRunnable(studyDto, exporter, csvWriter, activities, participants);
+            Thread csvExportThread = new Thread(csvExportRunnable);
+            csvExportThread.start();
+
+            String fileName = buildExportBlobFilename(studyDto);
+
+            // Google writing happens on this thread
+            return saveToGoogleBucket(csvInputStream, fileName, studyDto.getGuid(), bucket);
+        } catch (IOException e) {
+            throw new DDPException(e);
+        }
+    }
+
+    Runnable buildExportToCsvRunnable(StudyDto studyDto, DataExporter exporter, Writer csvOutputWriter,
+                                      List<ActivityExtract> activities,
+                                      Iterator<Participant> participants) {
+        return () -> {
+            try {
+                int total = exporter.exportDataSetAsCsv(studyDto, activities, participants, csvOutputWriter);
+                LOG.info("Written {} participants to csv export for study {}", total, studyDto.getGuid());
+                // closing here is important! Can't wait until the try block calls close
+                csvOutputWriter.close();
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        };
+    }
+
+    boolean saveToGoogleBucket(InputStream csvInputStream, String fileName, String studyGuid, Bucket bucket) {
+        Blob blob = bucket.create(fileName, csvInputStream, "text/csv");
+        LOG.info("Uploaded file {} to bucket {} for study {}", blob.getName(), bucket.getName(), studyGuid);
+        return true;
+    }
+
+    private String buildExportBlobFilename(StudyDto study) {
+        return String.format("%s/%s", study.getName(), DataExporter.makeExportCSVFilename(study.getGuid(), Instant.now()));
+    }
+
+    <R, X extends Exception> R withAPIsTxn(HandleCallback<R, X> callback) throws X {
+        return TransactionWrapper.withTxn(TransactionWrapper.DB.APIS, callback);
+    }
+
+    private class PaginatedParticipantIterator implements Iterator<Participant> {
```

Comment:
Given how the csv export is currently setup and how we're mocking data in `DataExporterTest`, I opted to keep csv export mostly the same and instead control the pagination from within the iterator.